lib/internal/modules/esm/module_job.js

```diff
@@ -2,26 +2,80 @@
 
 const {
   ArrayPrototypeJoin,
+  Number,
   ObjectSetPrototypeOf,
   PromiseAll,
   SafeSet,
   SafePromise,
   StringPrototypeIncludes,
   StringPrototypeMatch,
-  StringPrototypeReplace,
   StringPrototypeSplit,
 } = primordials;
 
 const { ModuleWrap } = internalBinding('module_wrap');
 
 const { decorateErrorStack } = require('internal/util');
+const { fileURLToPath } = require('url');
 const assert = require('internal/assert');
 const resolvedPromise = SafePromise.resolve();
 
 function noop() {}
 
 let hasPausedEntry = false;
 
+function extractExample(file, lineNumber) {
+  const { readFileSync } = require('fs');
+  const { parse } = require('internal/deps/acorn/acorn/dist/acorn');
+  const { findNodeAt } = require('internal/deps/acorn/acorn-walk/dist/walk');
+
+  const code = readFileSync(file, { encoding: 'utf8' });
+  const parsed = parse(code, {
```

Comment:
Is there any way around parsing the full file? I tried working with `parseExpressionAt` to only part the first few lines of a file but didn't have success (received `unexpected token` errors resulting from the `import` statements).